src/odemis/model/_futures.py

```diff
@@ -548,18 +551,18 @@ def _on_future_done(self, f):
         """
         self.set_progress(end=self._estimate_end())  # set the progress for batch future (all futures not done yet)
 
-        # Set exception if future failed and cancel all other sub-futures
-        try:
-            ex = f.exception()  # raises CancelledError if cancelled, otherwise returns error
-            if ex:
-                self.cancel()
-                self.set_exception(ex)
-                return
-        except CancelledError:
-            if self.cancel():  # if cancelling works
-                return
-            else:
-                self.set_exception(CancelledError())  # if cancelling fails
+        # If an exception occurs or CancelledError is raised for a sub-future, cancel the ProgressiveBatchFuture and
+        # all its sub-futures
+        if not self._is_cancel_triggered:
```

Comment:
I think it would be clearer to have an explicit "short-cut" like that:
```
if self._is_cancel_triggered:
    return
```
And then add a comment explaining why this code is needed.

Reply:
I will explain a bit more in the comment. The explicit "short-cut" cannot be used here because line 570 still needs to be called to later set the ProgressiveBatchFuture's result